packages/firestore/src/core/sync_engine.ts

```diff
@@ -270,7 +270,10 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
 
     const data = new QueryView(query, targetId, view);
     this.queryViewsByQuery.set(query, data);
-    this.queryViewsByTarget[targetId] = data;
+    if (!this.queriesByTarget[targetId]) {
+      this.queriesByTarget[targetId] = new Array<Query>();
```

Comment:
Optional Nit: I prefer `this.queriesByTarget[targetId] = []; `since it is more concise and used in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array

Reply:
Done.